src/dataflow/components/dataflow-program-topbar.tsx

```diff
@@ -12,43 +12,67 @@ interface TopbarProps {
   onProgramTimeSelectClick: (type: number) => void;
   isRunEnabled: boolean;
   runningProgram: boolean;
+  remainingTime: number;
   readOnly: boolean;
 }
 
 export const DataflowProgramTopbar = (props: TopbarProps) => {
   const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
     props.onProgramTimeSelectClick(Number(event.target.value));
   };
+  const runTime = props.programRunTimes.find( (rt: ProgramRunTime) => rt.val === props.programDefaultRunTime );
+  const remainingHours = Math.floor(props.remainingTime / 3600);
+  const remainingMinutes = Math.floor((props.remainingTime - remainingHours * 3600) / 60);
+  const remainingSeconds = (props.remainingTime - (remainingHours * 3600) - (remainingMinutes * 60) ) % 60;
+  const progressWidth = (76 * props.remainingTime / props.programDefaultRunTime).toString() + "px";
```

Comment:
Seems like the `76` here should be a named constant rather than a magic number for clarity.

Reply:
Fixed.